### ISSUE TITLE
Cannot find module

### ISSUE THREAD

**Issue by themre** [idyll-lang/idyll#20]
hi,
I did a fresh install of idyll and yo. Ran yo idyll and npm start and got:
```
Spellcheck:
var: did you mean vary
Error: Cannot find module 'c:\git\demo-idyll\c' from 'c:\git\demo-idyll'
```
anyone else got this problem?

**Comment by rreusser** [comment]
I wonder if maybe it's a windows path issue. To rule out the content and if it's not too long, can you post the `.idl` file? I've experienced it where syntax errors or other issues on my end cause messages that weren't necessarily what I needed to know to debug it.

**Comment by themre** [comment]
index.idl? I haven't changed it one bit. however when I open components.js, I see different slashes:
```
module.exports = {
"header": require('c:\git\demo-idyll\components\default/header'),
"range": require('c:\git\demo-idyll\components\default/range'),
"display-var": require('c:\git\demo-idyll\components\default/display-var'),
"custom-component": require('c:\git\demo-idyll\components/custom-component')
} 
```
perhaps this is the issue. I'll try with different slahes.

**Comment by themre** [comment]
nope, doesn't help, because it's generated one I run app. here is index.idl content:
```
[meta title:"Page Title" description:"Short description of your project" /]

[Header
  title:"Welcome to Idyll"
  subtitle:"Open index.idl to start writing"
  author:"Your Name Here"
  authorLink:"https://idyll-lang.github.io" /]


This is an Idyll file. Write text
as you please in here. To add interactivity,
you can add  different components to the text.

Here is how you can use a variable:

[var name:"exampleVar" value:5 /]

[Range min:0 max:10 value:exampleVar /]
[DisplayVar var:exampleVar /]

```
var code = true;
```

And here is a custom component:

[CustomComponent /]

You can use standard html tags if a
component with the same name
doesn't exist.

```

**Comment by rreusser** [comment]
Ah, apologies. It's become a knee-jerk reflex to ask or full input before debugging output, but I think the issue is [here](https://github.com/idyll-lang/idyll/blob/master/src/index.js#L138). Specifically that it creates paths by adding a `'/'` instead of the platform-agnostic `path.join`.

**Comment by themre** [comment]
any way to help debug this?

**Comment by mathisonian** [comment]
Thanks @rreusser, I'm guessing that's what it is. 

@themre I'll push a fix for the slashes today, but I don't have a windows machine to easily test on if you could take a look after that

**Comment by themre** [comment]
great. let me know and i'll test!

**Comment by rreusser** [comment]
See #20. I pushed… well, not so much a fix because I don't have a windows machine either, but it's at least a shot in the dark at replacing string-based path construction with `path` operations. Let me know if you think it's a good/bad move, @mathisonian.

@themre you might be able to clone, `npm install` and run `node bin/idyll.js` manually in the source directory in order to test, but maybe there's a better way to check this. Not sure. 🤷‍♂️

**Comment by themre** [comment]
tried, got this error:
```
fs.js:951
  return binding.readdir(pathModule._makeLong(path), options.encoding);
                 ^
Error: ENOENT: no such file or directory, scandir 'c:\git\idyll\components\default'
    at Error (native)
    at Object.fs.readdirSync (fs.js:951:18)
    at idyll (c:\git\idyll\src\index.js:63:25)
```

**Comment by rreusser** [comment]
I think that's progress. The subsequent path error is probably related to #20 + #21 and will require some debugging, I think.

**Comment by rreusser** [comment]
Thanks for testing! Hopefully that's progress. The line it points to is [this](https://github.com/idyll-lang/idyll/blob/2210c81b33d797ee5de84164bd92a191aa5dfef2/src/index.js#L63):

```javascript
const components = fs.readdirSync(DEFAULT_COMPONENTS_FOLDER);
```

I wonder if that needs to be something really explicit like `path.join(process.cwd(), DEFAULT_COMPONENTS_FOLDER)`.

**Comment by themre** [comment]
like `const components = path.join(process.cwd(), DEFAULT_COMPONENTS_FOLDER);` ?
if so, I get:
```
{ Error: ENOENT: no such file or directory, scandir 'c:\git\idyll\components'
    at Error (native)
    at Object.fs.readdirSync (fs.js:951:18)
    at idyll (c:\git\idyll\src\index.js:66:27)
    at Object.<anonymous> (c:\git\idyll\bin\idyll.js:38:1)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'c:\\git\\idyll\\components' }
fs.js:640
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

TypeError: path must be a string or Buffer
    at TypeError (native)
```

**Comment by mathisonian** [comment]
Yeah, I don't think `'c:\git\idyll\components\default'` is really the path we want to be scanning here. For some reason the logic constructing that isn't working correctly (maybe something with [this](https://github.com/idyll-lang/idyll/blob/2210c81b33d797ee5de84164bd92a191aa5dfef2/src/index.js#L29)?)

If it isn't obvious how to fix it, UW has some windows machines I can test on later today.

**Comment by rreusser** [comment]
Sorry, I meant 

```javascript
const components = fs.readdirSync(path.join(process.cwd(), DEFAULT_COMPONENTS_FOLDER));
```

but I think it might require a closer look at the relative requires. `path.join` strips out leading `'.'` so that you have to be a bit more careful. I think it probably needs  `__dirname` or `process.cwd()` in the right places, depending on whether it's relative to the idyll src directory or your project src directory.

**Comment by mathisonian** [comment]
I just merged #21, added automated testing in a windows environment, and released a patch on npm (`v1.1.2`). 

@themre let me know if you still have problems after this patch. To update either run `npm install idyll@1.1.2` inside of your idyll project folder, or create a new folder using the project generator.

**Comment by themre** [comment]
It works!
I had to upgrade also default-components:
```
"idyll": "~1.1.3",
"idyll-default-components": "^1.0.3"
```
Many thanks!
